### PR TITLE
fix: cheery-pick ingestion error logging

### PR DESCRIPTION
**Pull request by YashodhanJoshi1**
This cherry-picks the commits from https://github.com/openobserve/openobserve/pull/4903 for ingestion error logging.

## Release Notes

- **New Features**
  - Introduced a toggle for managing ingestion logs in the organization settings.
  - Added tracking for dropped records in usage statistics.

- **Enhancements**
  - Improved error handling and logging for ingestion processes across various components.
  - Enhanced metrics tracking for ingestion errors, including specific failure types.
  - Updated organization settings retrieval to include the new ingestion logs toggle.

- **Documentation**
  - Updated localization to include a label for the new ingestion logs toggle.

These changes collectively improve the user experience by providing better control over logging and enhancing the observability of the system.

### CODE REVIEW

File: src/service/logs/mod.rs [hunk 1]
```diff
@@ -509,6 +549,25 @@
     new_map
 }
 
+async fn ingestion_log_enabled() -> bool {
+    // the logging will be enabled through meta only, so hardcoded
+    match get_org_setting("_meta").await {
+        Ok(org_settings) => org_settings.toggle_ingestion_logs,
+        Err(_) => false,
+    }
+}
```
[the remark this answers is not in the document]
Comment (YashodhanJoshi1) [reply src/service/logs/mod.rs:558]: this is done intentionally, there is a comment in the code for this.

File: src/service/logs/mod.rs [hunk 2]
```diff
@@ -509,6 +549,25 @@ pub fn refactor_map(
...
+fn log_failed_record<T: std::fmt::Debug>(enabled: bool, record: &T, error: &str) {
+    if !enabled {
+        return;
+    }
+    log::warn!(
+        "failed to process record with error {} : {:?} ",
+        error,
+        record
+    );
+}
```
[the remark this answers is not in the document]
Comment (YashodhanJoshi1) [reply src/service/logs/mod.rs:569]: This is a requirement.

File: src/service/schema.rs [hunk 3]
```diff
@@ -96,6 +98,14 @@ pub async fn check_for_schema(
...
 
     if inferred_schema.fields.len() > cfg.limit.req_cols_per_record_limit {
+        metrics::INGEST_ERRORS
+            .with_label_values(&[
+                org_id,
+                StreamType::Logs.to_string().as_str(),
+                stream_name,
+                SCHEMA_CONFORMANCE_FAILED,
+            ])
+            .inc();
```
[the remark this answers is not in the document]
Comment (YashodhanJoshi1) [reply src/service/schema.rs:108]: fixed

File: src/config/src/meta/usage.rs [hunk 4]
```diff
@@ -86,6 +86,8 @@ pub struct UsageData {
     pub response_time: f64,
     pub stream_type: StreamType,
     pub num_records: i64,
+    #[serde[default]]
+    pub dropped_records: i64,
```
[the remark this answers is not in the document]
Comment (YashodhanJoshi1) [reply src/config/src/meta/usage.rs:90]: fixed

File: src/service/usage/mod.rs [hunk 5]
```diff
@@ -279,6 +281,7 @@ async fn ingest_usages(curr_usages: Vec<UsageData>) {
             continue;
         } else {
             entry.usage_data.num_records += usage_data.num_records;
+            entry.usage_data.dropped_records += usage_data.dropped_records;
             entry.usage_data.size += usage_data.size;
             entry.usage_data.response_time += usage_data.response_time;
             entry.count += 1;
```
[the remark this answers is not in the document]
Comment (YashodhanJoshi1) [reply src/service/usage/mod.rs:287]: This is existing code, the is_new is not a new addition.

**Review by hengfeiyang — APPROVED** [review]

**Pull request merged**